File: CS/ViewModel.cs
using DevExpress.Mvvm;
using System.Collections.ObjectModel;

namespace E20031 {
    public class TestViewModel : BindableBase {
        ObservableCollection<TestData> _records;
        public ObservableCollection<TestData> Records {
            get { return _records; }
            set {
                if (_records != value) {
                    _records = value;
                    RaisePropertyChanged(() => Records);
                }
            }
        }
        public TestViewModel() {
            Records = new ObservableCollection<TestData>();
            for (int i = 0; i < 10; i++)
                Records.Add(new TestData() { Text = "Row" + i, Number = i });
        }
    }
    public class TestData {
        public string Text { get; set; }
        public int Number { get; set; }
    }
}

File: VB/ViewModel.vb
Imports DevExpress.Mvvm
Imports System.Collections.ObjectModel

Namespace E20031

    Public Class TestViewModel
        Inherits BindableBase

        Private _records As ObservableCollection(Of TestData)

        Public Property Records As ObservableCollection(Of TestData)
            Get
                Return _records
            End Get

            Set(ByVal value As ObservableCollection(Of TestData))
                If _records IsNot value Then
                    _records = value
                    RaisePropertyChanged(Function() Records)
                End If
            End Set
        End Property

        Public Sub New()
            Records = New ObservableCollection(Of TestData)()
            For i As Integer = 0 To 10 - 1
                Records.Add(New TestData() With {.Text = "Row" & i, .Number = i})
            Next
        End Sub
    End Class

    Public Class TestData

        Public Property Text As String

        Public Property Number As Integer
    End Class
End Namespace
